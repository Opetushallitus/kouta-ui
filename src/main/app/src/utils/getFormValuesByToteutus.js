import get from 'lodash/get';

const getFormValuesByToteutus = toteutus => {
  const {
    kielivalinta = [],
    nimi = {},
    tarjoajat = [],
    metadata = {},
  } = toteutus;

  const {
    kuvaus = {},
    ammattinimikkeet = [],
    asiasanat = [],
    opetus = {},
    osaamisalat: osaamisalatArg = [],
    ylemmanKorkeakoulututkinnonOsaamisalat: ylemmanKorkeakoulututkinnonOsaamisalatArg = [],
    alemmanKorkeakoulututkinnonOsaamisalat: alemmanKorkeakoulututkinnonOsaamisalatArg = [],
    yhteyshenkilot = [],
    diplomiKoodiUrit,
    diplomiKuvaus,
    A1JaA2Kielivalikoima,
    aidinkieliKielivalikoima,
    B1Kielivalikoima,
    B2Kielivalikoima,
    B3Kielivalikoima,
    muuKielivalikoima,
    lukiolinjaKoodiUri,
    jaksonKuvaus,
  } = metadata;

  const osaamisalat = osaamisalatArg.map(({ koodi }) => koodi);

  const { osaamisalaLinkit, osaamisalaLinkkiOtsikot } = osaamisalatArg.reduce(
    (acc, curr) => {
      const { koodi, linkki = {}, otsikko = {} } = curr;

      if (koodi) {
        acc.osaamisalaLinkit[koodi] = linkki;
        acc.osaamisalaLinkkiOtsikot[koodi] = otsikko;
      }

      return acc;
    },
    { osaamisalaLinkit: {}, osaamisalaLinkkiOtsikot: {} },
  );

  const { lisatiedot = [] } = opetus;

  const osiot = lisatiedot
    .filter(({ otsikkoKoodiUri }) => !!otsikkoKoodiUri)
    .map(({ otsikkoKoodiUri }) => ({ value: otsikkoKoodiUri }));

  const osioKuvaukset = lisatiedot.reduce((acc, curr) => {
    if (curr.otsikkoKoodiUri) {
      acc[curr.otsikkoKoodiUri] = curr.teksti || {};
    }

    return acc;
  }, {});

  const maksullistyyppi = get(opetus, 'onkoLukuvuosimaksua')
    ? 'lukuvuosimaksu'
    : get(opetus, 'onkoMaksullinen')
    ? 'kylla'
    : 'ei';

  return {
    nimi: nimi,
    kieliversiot: kielivalinta,
    jarjestamispaikat: tarjoajat,
    jarjestamistiedot: {
      kuvaus,
      maksullisuus: {
        tyyppi: maksullistyyppi,
        maksu: (maksullistyyppi === 'lukuvuosimaksu'
          ? get(opetus, 'lukuvuosimaksu') || ''
          : get(opetus, 'maksunMaara') || ''
        ).toString(),
      },
      maksumaara: get(opetus, 'maksunMaara') || {},
      opetustapa: get(opetus, 'opetustapaKoodiUrit') || [],
      opetusaika: get(opetus, 'opetusaikaKoodiUrit') || [],
      opetuskieli: get(opetus, 'opetuskieliKoodiUrit') || [],
      opetusaikaKuvaus: get(opetus, 'opetusaikaKuvaus') || {},
      opetustapaKuvaus: get(opetus, 'opetustapaKuvaus') || {},
      opetuskieliKuvaus: get(opetus, 'opetuskieletKuvaus') || {},
      maksullisuusKuvaus: get(opetus, 'maksullisuusKuvaus') || {},
      osiot,
      osioKuvaukset,
      alkamiskausi: {
        kausi: get(opetus, 'alkamiskausiKoodiUri') || '',
        vuosi: { value: get(opetus, 'alkamisvuosi') || '' },
      },
      alkamiskausiKuvaus: get(opetus, 'alkamisaikaKuvaus') || {},
      onkoLukuvuosimaksua: get(opetus, 'onkoLukuvuosimaksua'),
      lukuvuosimaksu: get(opetus, 'lukuvuosimaksu') || {},
      lukuvuosimaksuKuvaus: get(opetus, 'lukuvuosimaksuKuvaus') || {},
      onkoStipendia: get(opetus, 'onkoStipendia'),
      stipendinMaara: get(opetus, 'stipendinMaara') || {},
      stipendinKuvaus: get(opetus, 'stipendinKuvaus') || {},
      diplomiTyypit: (diplomiKoodiUrit || []).map(value => ({ value })),
      diplomiKuvaus: diplomiKuvaus || {},
      A1A2Kielet: (A1JaA2Kielivalikoima || []).map(value => ({ value })),
      aidinkielet: (aidinkieliKielivalikoima || []).map(value => ({
        value,
      })),
      B1Kielet: (B1Kielivalikoima || []).map(value => ({ value })),
      B2Kielet: (B2Kielivalikoima || []).map(value => ({ value })),
      B3Kielet: (B3Kielivalikoima || []).map(value => ({ value })),
      muutKielet: (muuKielivalikoima || []).map(value => ({ value })),
    },
    nayttamistiedot: {
      ammattinimikkeet: ammattinimikkeet.reduce((acc, curr) => {
        if (get(curr, 'kieli') && get(curr, 'arvo')) {
          acc[curr.kieli] = acc[curr.kieli] || [];
          acc[curr.kieli] = [
            ...acc[curr.kieli],
            { label: curr.arvo, value: curr.arvo },
          ];
        }

        return acc;
      }, {}),
      avainsanat: asiasanat.reduce((acc, curr) => {
        if (get(curr, 'kieli') && get(curr, 'arvo')) {
          acc[curr.kieli] = acc[curr.kieli] || [];
          acc[curr.kieli] = [
            ...acc[curr.kieli],
            { label: curr.arvo, value: curr.arvo },
          ];
        }

        return acc;
      }, {}),
    },
    yhteyshenkilot: yhteyshenkilot.map(
      ({ nimi, titteli, sahkoposti, wwwSivu, puhelinnumero }) => ({
        nimi: nimi || {},
        titteli: titteli || {},
        sahkoposti: sahkoposti || {},
        verkkosivu: wwwSivu || {},
        puhelinnumero: puhelinnumero || {},
      }),
    ),
    osaamisalat: {
      osaamisalat,
      osaamisalaLinkit,
      osaamisalaLinkkiOtsikot,
    },
    ylemmanKorkeakoulututkinnonOsaamisalat: ylemmanKorkeakoulututkinnonOsaamisalatArg,
    alemmanKorkeakoulututkinnonOsaamisalat: alemmanKorkeakoulututkinnonOsaamisalatArg,
    kuvaus,
    lukiolinjat: {
      linja: lukiolinjaKoodiUri ? { value: lukiolinjaKoodiUri } : '',
      jaksonKuvaus: jaksonKuvaus || {},
    },
  };
};

export default getFormValuesByToteutus;
