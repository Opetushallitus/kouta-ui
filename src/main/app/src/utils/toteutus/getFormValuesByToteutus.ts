import _fp from 'lodash/fp';

import { parseEditorState } from '#/src/components/Editor/utils';
import { ApurahaMaaraTyyppi, ApurahaYksikko } from '#/src/constants';
import {
  ToteutusFormValues,
  MaksullisuusTyyppi,
} from '#/src/types/toteutusTypes';
import { toSelectValue, toSelectValueList } from '#/src/utils';
import { getAjankohtaFields } from '#/src/utils/form/aloitusajankohtaHelpers';
import parseSisaltoField from '#/src/utils/form/parseSisaltoField';

const kieliArvoListToMultiSelectValue = _fp.reduce((acc, curr: any) => {
  if (curr?.kieli && curr?.arvo) {
    return {
      ...acc,
      [curr.kieli]: [
        ...(acc[curr.kieli] ?? []),
        { label: curr.arvo, value: curr.arvo },
      ],
    };
  }

  return acc;
}, {});

const getFormValuesByToteutus = (toteutus): ToteutusFormValues => {
  const {
    koulutustyyppi,
    kielivalinta,
    nimi,
    tarjoajat,
    metadata = {},
    tila,
    teemakuva,
    sorakuvausId,
    esikatselu = false,
  } = toteutus;

  const {
    kuvaus = {},
    ammattinimikkeet,
    asiasanat,
    opetus = {},
    osaamisalat,
    ylemmanKorkeakoulututkinnonOsaamisalat,
    alemmanKorkeakoulututkinnonOsaamisalat,
    yhteyshenkilot,
    lukiolinjaKoodiUri,
    laajuus,
    laajuusyksikkoKoodiUri,
    ilmoittautumislinkki,
    aloituspaikat,
    toteutusjaksot,
    ammatillinenPerustutkintoErityisopetuksena,
  } = metadata;

  const {
    diplomiKoodiUrit,
    diplomiKuvaus,
    A1JaA2Kielivalikoima,
    aidinkieliKielivalikoima,
    B1Kielivalikoima,
    B2Kielivalikoima,
    B3Kielivalikoima,
    muuKielivalikoima,
    lisatiedot,
    koulutuksenAlkamiskausi = {},
    maksullisuustyyppi = MaksullisuusTyyppi.MAKSUTON,
  } = opetus;

  const { osaamisalaLinkit, osaamisalaLinkkiOtsikot } = _fp.reduce(
    (acc, curr: any) => {
      const { koodiUri, linkki = {}, otsikko = {} } = curr;

      if (koodiUri) {
        acc.osaamisalaLinkit[koodiUri] = linkki;
        acc.osaamisalaLinkkiOtsikot[koodiUri] = otsikko;
      }

      return acc;
    },
    { osaamisalaLinkit: {}, osaamisalaLinkkiOtsikot: {} }
  )(osaamisalat);

  return {
    koulutustyyppi,
    tila,
    esikatselu,
    tiedot: {
      nimi: nimi ?? {},
      ammatillinenPerustutkintoErityisopetuksena: Boolean(
        ammatillinenPerustutkintoErityisopetuksena
      ),
      laajuus: _fp.isNumber(laajuus) ? laajuus.toString() : '',
      laajuusyksikko: toSelectValue(laajuusyksikkoKoodiUri),
      ilmoittautumislinkki: ilmoittautumislinkki || {},
      aloituspaikat: _fp.isNumber(aloituspaikat)
        ? aloituspaikat.toString()
        : '',
    },
    kuvaus: _fp.mapValues(parseEditorState, kuvaus || {}),
    kieliversiot: kielivalinta ?? [],
    tarjoajat: tarjoajat ?? [],
    jarjestamistiedot: {
      maksullisuustyyppi,
      maksunMaara: opetus?.maksunMaara,
      opetustapa: opetus?.opetustapaKoodiUrit || [],
      opetusaika: opetus?.opetusaikaKoodiUrit || [],
      opetuskieli: opetus?.opetuskieliKoodiUrit || [],
      suunniteltuKestoKuvaus: _fp.mapValues(
        parseEditorState,
        opetus?.suunniteltuKestoKuvaus || {}
      ),
      suunniteltuKesto: {
        vuotta: opetus?.suunniteltuKestoVuodet,
        kuukautta: opetus?.suunniteltuKestoKuukaudet,
      },
      opetusaikaKuvaus: _fp.mapValues(
        parseEditorState,
        opetus?.opetusaikaKuvaus || {}
      ),
      opetustapaKuvaus: _fp.mapValues(
        parseEditorState,
        opetus?.opetustapaKuvaus || {}
      ),
      opetuskieliKuvaus: _fp.mapValues(
        parseEditorState,
        opetus?.opetuskieletKuvaus || {}
      ),
      maksullisuusKuvaus: _fp.mapValues(
        parseEditorState,
        opetus?.maksullisuusKuvaus || {}
      ),
      osiot: _fp.flow(
        _fp.filter(({ otsikkoKoodiUri }) => !!otsikkoKoodiUri),
        _fp.map(({ otsikkoKoodiUri }) => ({ value: otsikkoKoodiUri }))
      )(lisatiedot),
      osioKuvaukset: _fp.reduce((acc, curr: any) => {
        if (curr.otsikkoKoodiUri) {
          acc[curr.otsikkoKoodiUri] = _fp.mapValues(
            parseEditorState,
            curr.teksti || {}
          );
        }
        return acc;
      }, {})(lisatiedot),
      onkoApuraha: opetus?.onkoApuraha,
      apurahaMin: opetus?.apuraha?.min,
      apurahaMax: opetus?.apuraha?.max,
      apurahaMaaraTyyppi:
        opetus?.apuraha?.min === opetus?.apuraha?.max
          ? ApurahaMaaraTyyppi.YKSI_ARVO
          : ApurahaMaaraTyyppi.VAIHTELUVALI,
      apurahaYksikko: toSelectValue(
        opetus?.apuraha?.yksikko ?? ApurahaYksikko.EURO
      ),
      apurahaKuvaus: _fp.mapValues(
        parseEditorState,
        opetus?.apuraha?.kuvaus || {}
      ),
      diplomiTyypit: toSelectValueList(diplomiKoodiUrit),
      diplomiKuvaus: _fp.mapValues(parseEditorState, diplomiKuvaus ?? {}),
      A1A2Kielet: toSelectValueList(A1JaA2Kielivalikoima),
      aidinkielet: toSelectValueList(aidinkieliKielivalikoima),
      B1Kielet: toSelectValueList(B1Kielivalikoima),
      B2Kielet: toSelectValueList(B2Kielivalikoima),
      B3Kielet: toSelectValueList(B3Kielivalikoima),
      muutKielet: toSelectValueList(muuKielivalikoima),
      ajankohta: getAjankohtaFields(koulutuksenAlkamiskausi),
    },
    nayttamistiedot: {
      ammattinimikkeet: kieliArvoListToMultiSelectValue(ammattinimikkeet),
      avainsanat: kieliArvoListToMultiSelectValue(asiasanat),
    },
    yhteyshenkilot: _fp.map(
      ({ nimi, titteli, sahkoposti, wwwSivu, puhelinnumero }) => ({
        nimi: nimi || {},
        titteli: titteli || {},
        sahkoposti: sahkoposti || {},
        verkkosivu: wwwSivu || {},
        puhelinnumero: puhelinnumero || {},
      })
    )(yhteyshenkilot),
    osaamisalat: {
      osaamisalat: _fp.map(({ koodiUri }) => koodiUri)(osaamisalat),
      osaamisalaLinkit,
      osaamisalaLinkkiOtsikot,
    },
    ylemmanKorkeakoulututkinnonOsaamisalat:
      _fp.map(({ kuvaus, nimi, linkki, otsikko }) => ({
        kuvaus: _fp.mapValues(parseEditorState, kuvaus),
        nimi,
        linkki,
        otsikko,
      }))(ylemmanKorkeakoulututkinnonOsaamisalat) ?? [],
    alemmanKorkeakoulututkinnonOsaamisalat:
      _fp.map(({ kuvaus, nimi, linkki, otsikko }) => ({
        kuvaus: _fp.mapValues(parseEditorState, kuvaus),
        nimi,
        linkki,
        otsikko,
      }))(alemmanKorkeakoulututkinnonOsaamisalat) ?? [],
    lukiolinjat: {
      lukiolinja: lukiolinjaKoodiUri
        ? { value: lukiolinjaKoodiUri }
        : undefined,
    },
    toteutusjaksot: _fp.map(
      ({
        nimi = {},
        koodi = '',
        laajuus = {},
        ilmoittautumislinkki = {},
        kuvaus = {},
        sisalto,
      }) => ({
        nimi,
        koodi,
        laajuus,
        ilmoittautumislinkki,
        kuvaus: _fp.mapValues(parseEditorState, kuvaus),
        sisalto: parseSisaltoField(sisalto),
      })
    )(toteutusjaksot),
    teemakuva,
    hakeutumisTaiIlmoittautumistapa: {
      hakeutumisTaiIlmoittautumistapa: metadata?.hakulomaketyyppi,
      hakuTapa: metadata?.hakutermi,
      linkki: metadata?.hakulomakeLinkki,
      lisatiedot: _fp.mapValues(
        parseEditorState,
        metadata?.lisatietoaHakeutumisesta
      ),
      lisatiedotValintaperusteista: _fp.mapValues(
        parseEditorState,
        metadata?.lisatietoaValintaperusteista
      ),
      hakuaikaAlkaa: metadata?.hakuaika?.alkaa,
      hakuaikaPaattyy: metadata?.hakuaika?.paattyy,
    },
    soraKuvaus: toSelectValue(sorakuvausId),
  };
};

export default getFormValuesByToteutus;
