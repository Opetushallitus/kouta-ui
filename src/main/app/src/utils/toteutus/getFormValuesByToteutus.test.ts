import { Alkamiskausityyppi, JULKAISUTILA } from '#/src/constants';
import getFormValuesByToteutus from '#/src/utils/toteutus/getFormValuesByToteutus';

test('getFormValuesByToteutus returns correct form values given toteutus', () => {
  const values = getFormValuesByToteutus({
    tila: JULKAISUTILA.TALLENNETTU,
    kielivalinta: ['fi', 'sv'],
    metadata: {
      alemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: {
            fi: 'Fi alempikuvaus',
            sv: 'Sv alempikuvaus',
          },
          linkki: {
            fi: 'Fi alempilinkki',
            sv: 'Sv alempilinkki',
          },
          nimi: {
            fi: 'Fi alempinimi',
            sv: 'Sv alempinimi',
          },
          otsikko: {
            fi: 'Fi alempiotsikko',
            sv: 'Sv alempiotsikko',
          },
        },
      ],
      ammattinimikkeet: [
        {
          arvo: 'nimike1',
          kieli: 'fi',
        },
        {
          arvo: 'nimike2',
          kieli: 'sv',
        },
      ],
      asiasanat: [
        {
          arvo: 'avainsana1',
          kieli: 'fi',
        },
        {
          arvo: 'avainsana2',
          kieli: 'sv',
        },
      ],
      kuvaus: {
        fi: 'Fi toteutuksenkuvaus',
        sv: 'Sv toteutuksenkuvaus',
      },
      opetus: {
        koulutuksenAlkamiskausi: {
          alkamiskausityyppi: Alkamiskausityyppi.TARKKA_ALKAMISAJANKOHTA,
          koulutuksenAlkamiskausiKoodiUri: 'kausi_0#1',
          koulutuksenAlkamispaivamaara: '2021-04-16T00:00',
          kouutuksenPaattymispaivamaara: '2021-12-12T00:00',
          koulutuksenAlkamisvuosi: 2020,
        },
        lisatiedot: [
          {
            otsikkoKoodiUri: 'osio_1#1',
            teksti: {
              fi: 'Fi kuvaus1',
              sv: 'Sv kuvaus1',
            },
          },
          {
            otsikkoKoodiUri: 'osio_2#1',
            teksti: {
              fi: 'Fi kuvaus2',
              sv: 'Sv kuvaus2',
            },
          },
        ],
        maksullisuus: {
          maksu: '',
        },
        maksumaara: {},
        maksullisuusKuvaus: {
          fi: 'Fi maksullisuuskuvaus',
          sv: 'Sv maksullisuuskuvaus',
        },
        onkoMaksullinen: true,
        onkoStipendia: true,
        suunniteltuKestoKuvaus: {
          fi: 'Fi suunniteltuKestoKuvaus',
          sv: 'Sv suunniteltuKestoKuvaus',
        },
        suunniteltuKestoVuodet: 2,
        suunniteltuKestoKuukaudet: 6,
        opetusaikaKoodiUrit: ['opetusaika_1#1'],
        opetusaikaKuvaus: {
          fi: 'Fi aikakuvaus',
          sv: 'Sv aikakuvaus',
        },
        opetuskieletKuvaus: {
          fi: 'Fi kielikuvaus',
          sv: 'Sv kielikuvaus',
        },
        opetuskieliKoodiUrit: ['opetuskieli_1#1', 'opetuskieli_2#1'],
        opetustapaKoodiUrit: ['opetustapa_1#1', 'opetustapa_2#1'],
        opetustapaKuvaus: {
          fi: 'Fi tapakuvaus',
          sv: 'Sv tapakuvaus',
        },
        stipendinKuvaus: {
          fi: 'Fi stipendikuvaus',
          sv: 'Sv stipendikuvaus',
        },
        stipendinMaara: 20,
        diplomiKoodiUrit: ['dipmlomi_1#1'],
        diplomiKuvaus: {
          fi: 'Fi diplomi',
          sv: 'Sv diplomi',
        },
        A1JaA2Kielivalikoima: ['kieli_1#1'],
        B2Kielivalikoima: ['kieli_2#1'],
        B1Kielivalikoima: ['kieli_3#1'],
        B3Kielivalikoima: ['kieli_4#1'],
        aidinkieliKielivalikoima: ['kieli_5#1'],
        muuKielivalikoima: ['kieli_6#1'],
      },
      osaamisalat: [
        {
          koodiUri: 'osaamisala1',
          linkki: {
            fi: 'Fi linkki1',
            sv: 'Sv linkki1',
          },
          otsikko: {
            fi: 'Fi otsikko1',
            sv: 'Sv otsikko1',
          },
        },
        {
          koodiUri: 'osaamisala2',
          linkki: {
            fi: 'Fi linkki2',
            sv: 'Sv linkki2',
          },
          otsikko: {
            fi: 'Fi otsikko2',
            sv: 'Sv otsikko2',
          },
        },
      ],
      yhteyshenkilot: [
        {
          nimi: {
            fi: 'Fi nimi',
            sv: 'Sv nimi',
          },
          puhelinnumero: {
            fi: 'Fi puhelinnumero',
            sv: 'Sv puhelinnumero',
          },
          sahkoposti: {
            fi: 'Fi sähköposti',
            sv: 'Sv sähköposti',
          },
          titteli: {
            fi: 'Fi titteli',
            sv: 'Sv titteli',
          },
          wwwSivu: {
            fi: 'Fi verkkosivu',
            sv: 'Sv verkkosivu',
          },
        },
      ],
      ylemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: {
            fi: 'Fi ylempikuvaus',
            sv: 'Sv ylempikuvaus',
          },
          linkki: {
            fi: 'Fi ylempilinkki',
            sv: 'Sv ylempilinkki',
          },
          nimi: {
            fi: 'Fi ylempinimi',
            sv: 'Sv ylempinimi',
          },
          otsikko: {
            fi: 'Fi ylempiotsikko',
            sv: 'Sv ylempiotsikko',
          },
        },
      ],
      lukiolinjaKoodiUri: 'linja_1#1',
      tutkinnonOsat: [
        {
          osaamisalaKoodiUri: 'osaamisala_1#1',
          tutkinnonOsaKoodiUrit: ['tutkinnonosa_1#1'],
          tutkintoKoodiUri: 'tutkinto_1#1',
        },
      ],
      laajuus: 25,
      laajuusyksikkoKoodiUri: 'laajuus_1#1',
      ilmoittautumislinkki: {
        fi: 'Fi linkki',
        sv: 'Sv linkki',
      },
      aloituspaikat: 251,
      suunniteltuKesto: {
        fi: 'Fi kesto',
        sv: 'Sv kesto',
      },
      toteutusjaksot: [
        {
          ilmoittautumislinkki: {
            fi: 'Fi linkki',
            sv: 'Sv linkki',
          },
          koodi: 'koodi 123',
          kuvaus: {
            fi: 'Fi kuvaus',
            sv: 'Sv kuvaus',
          },
          laajuus: {
            fi: 'Fi laajuus',
            sv: 'Sv laajuus',
          },
          nimi: {
            fi: 'Fi nimi',
            sv: 'Sv nimi',
          },
          sisalto: [
            {
              data: {
                fi: '<h2>Fi sisalto</h2>',
                sv: '<h2>Sv sisalto</h2>',
              },
              tyyppi: 'teksti',
            },
            {
              data: {
                rows: [
                  {
                    columns: [
                      {
                        index: 0,
                        text: {
                          fi: 'Fi column1',
                          sv: 'Sv column1',
                        },
                      },
                      {
                        index: 1,
                        text: {
                          fi: 'Fi column2',
                          sv: 'Sv column2',
                        },
                      },
                    ],
                    index: 0,
                  },
                  {
                    columns: [
                      {
                        index: 0,
                        text: {
                          fi: 'Fi column3',
                          sv: 'Sv column3',
                        },
                      },
                      {
                        index: 1,
                        text: {
                          fi: 'Fi column4',
                          sv: 'Sv column4',
                        },
                      },
                    ],
                    index: 1,
                  },
                ],
              },
              tyyppi: 'taulukko',
            },
          ],
        },
      ],
    },
    nimi: {
      fi: 'Fi nimi',
      sv: 'Sv nimi',
    },
    kuvaus: {
      fi: 'Fi toteutuksenkuvaus',
      sv: 'Sv toteutuksenkuvaus',
    },
    tarjoajat: ['org1', 'org2'],
  });

  expect(values).toMatchSnapshot();
});
