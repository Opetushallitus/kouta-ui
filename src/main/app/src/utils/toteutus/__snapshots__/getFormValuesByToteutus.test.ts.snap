// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFormValuesByToteutus returns correct form values given toteutus 1`] = `
Object {
  "alemmanKorkeakoulututkinnonOsaamisalat": Array [
    Object {
      "kuvaus": Object {
        "fi": "<p>Fi alempikuvaus</p>",
        "sv": "<p>Sv alempikuvaus</p>",
      },
      "linkki": Object {
        "fi": "Fi alempilinkki",
        "sv": "Sv alempilinkki",
      },
      "nimi": Object {
        "fi": "Fi alempinimi",
        "sv": "Sv alempinimi",
      },
      "otsikko": Object {
        "fi": "Fi alempiotsikko",
        "sv": "Sv alempiotsikko",
      },
    },
  ],
  "esikatselu": false,
  "hakeutumisTaiIlmoittautumistapa": Object {
    "hakeutumisTaiIlmoittautumistapa": undefined,
    "hakuTapa": undefined,
    "hakuaikaAlkaa": undefined,
    "hakuaikaPaattyy": undefined,
    "linkki": undefined,
    "lisatiedot": Object {},
    "lisatiedotValintaperusteista": Object {},
  },
  "jarjestamistiedot": Object {
    "A1A2Kielet": Array [
      Object {
        "value": "kieli_1#1",
      },
    ],
    "B1Kielet": Array [
      Object {
        "value": "kieli_3#1",
      },
    ],
    "B2Kielet": Array [
      Object {
        "value": "kieli_2#1",
      },
    ],
    "B3Kielet": Array [
      Object {
        "value": "kieli_4#1",
      },
    ],
    "aidinkielet": Array [
      Object {
        "value": "kieli_5#1",
      },
    ],
    "ajankohta": Object {
      "ajankohtaTyyppi": "tarkka alkamisajankohta",
      "henkilokohtaisenSuunnitelmanLisatiedot": Object {},
      "kausi": "kausi_k#1",
      "tarkkaAlkaa": "2021-04-16T00:00",
      "tarkkaPaattyy": "2021-12-12T00:00",
      "vuosi": Object {
        "value": 2020,
      },
    },
    "apurahaKuvaus": Object {
      "fi": "<p>Fi apurahakuvaus</p>",
      "sv": "<p>Sv apurahakuvaus</p>",
    },
    "apurahaMaaraTyyppi": Object {
      "value": "range",
    },
    "apurahaMax": 200,
    "apurahaMin": 100,
    "apurahaTyyppi": "apuraha",
    "apurahaYksikko": Object {
      "value": "euroa",
    },
    "diplomiKuvaus": Object {
      "fi": "<p>Fi diplomi</p>",
      "sv": "<p>Sv diplomi</p>",
    },
    "diplomiTyypit": Array [
      Object {
        "value": "dipmlomi_1#1",
      },
    ],
    "maksullisuus": Object {
      "maksu": "",
      "tyyppi": "kylla",
    },
    "maksullisuusKuvaus": Object {
      "fi": "<p>Fi maksullisuuskuvaus</p>",
      "sv": "<p>Sv maksullisuuskuvaus</p>",
    },
    "maksumaara": Object {},
    "muutKielet": Array [
      Object {
        "value": "kieli_6#1",
      },
    ],
    "opetusaika": Array [
      "opetusaika_1#1",
    ],
    "opetusaikaKuvaus": Object {
      "fi": "<p>Fi aikakuvaus</p>",
      "sv": "<p>Sv aikakuvaus</p>",
    },
    "opetuskieli": Array [
      "opetuskieli_1#1",
      "opetuskieli_2#1",
    ],
    "opetuskieliKuvaus": Object {
      "fi": "<p>Fi kielikuvaus</p>",
      "sv": "<p>Sv kielikuvaus</p>",
    },
    "opetustapa": Array [
      "opetustapa_1#1",
      "opetustapa_2#1",
    ],
    "opetustapaKuvaus": Object {
      "fi": "<p>Fi tapakuvaus</p>",
      "sv": "<p>Sv tapakuvaus</p>",
    },
    "osioKuvaukset": Object {
      "osio_1#1": Object {
        "fi": "<p>Fi kuvaus1</p>",
        "sv": "<p>Sv kuvaus1</p>",
      },
      "osio_2#1": Object {
        "fi": "<p>Fi kuvaus2</p>",
        "sv": "<p>Sv kuvaus2</p>",
      },
    },
    "osiot": Array [
      Object {
        "value": "osio_1#1",
      },
      Object {
        "value": "osio_2#1",
      },
    ],
    "suunniteltuKesto": Object {
      "kuukautta": 6,
      "vuotta": 2,
    },
    "suunniteltuKestoKuvaus": Object {
      "fi": "<p>Fi suunniteltuKestoKuvaus</p>",
      "sv": "<p>Sv suunniteltuKestoKuvaus</p>",
    },
  },
  "kieliversiot": Array [
    "fi",
    "sv",
  ],
  "koulutustyyppi": undefined,
  "kuvaus": Object {
    "fi": "<p>Fi toteutuksenkuvaus</p>",
    "sv": "<p>Sv toteutuksenkuvaus</p>",
  },
  "lukiolinjat": Object {
    "lukiolinja": Object {
      "value": "linja_1#1",
    },
  },
  "nayttamistiedot": Object {
    "ammattinimikkeet": Object {
      "fi": Array [
        Object {
          "label": "nimike1",
          "value": "nimike1",
        },
      ],
      "sv": Array [
        Object {
          "label": "nimike2",
          "value": "nimike2",
        },
      ],
    },
    "avainsanat": Object {
      "fi": Array [
        Object {
          "label": "avainsana1",
          "value": "avainsana1",
        },
      ],
      "sv": Array [
        Object {
          "label": "avainsana2",
          "value": "avainsana2",
        },
      ],
    },
  },
  "osaamisalat": Object {
    "osaamisalaLinkit": Object {
      "osaamisala1": Object {
        "fi": "Fi linkki1",
        "sv": "Sv linkki1",
      },
      "osaamisala2": Object {
        "fi": "Fi linkki2",
        "sv": "Sv linkki2",
      },
    },
    "osaamisalaLinkkiOtsikot": Object {
      "osaamisala1": Object {
        "fi": "Fi otsikko1",
        "sv": "Sv otsikko1",
      },
      "osaamisala2": Object {
        "fi": "Fi otsikko2",
        "sv": "Sv otsikko2",
      },
    },
    "osaamisalat": Array [
      "osaamisala1",
      "osaamisala2",
    ],
  },
  "soraKuvaus": undefined,
  "tarjoajat": Array [
    "org1",
    "org2",
  ],
  "teemakuva": undefined,
  "tiedot": Object {
    "aloituspaikat": "251",
    "ammatillinenPerustutkintoErityisopetuksena": true,
    "ilmoittautumislinkki": Object {
      "fi": "Fi linkki",
      "sv": "Sv linkki",
    },
    "laajuus": "25",
    "laajuusyksikko": Object {
      "value": "laajuus_1#1",
    },
    "nimi": Object {
      "fi": "Fi nimi",
      "sv": "Sv nimi",
    },
  },
  "tila": "tallennettu",
  "toteutusjaksot": Array [
    Object {
      "ilmoittautumislinkki": Object {
        "fi": "Fi linkki",
        "sv": "Sv linkki",
      },
      "koodi": "koodi 123",
      "kuvaus": Object {
        "fi": "<p>Fi kuvaus</p>",
        "sv": "<p>Sv kuvaus</p>",
      },
      "laajuus": Object {
        "fi": "Fi laajuus",
        "sv": "Sv laajuus",
      },
      "nimi": Object {
        "fi": "Fi nimi",
        "sv": "Sv nimi",
      },
      "sisalto": Array [
        Object {
          "data": Object {
            "fi": "<h2>Fi sisalto</h2>",
            "sv": "<h2>Sv sisalto</h2>",
          },
          "tyyppi": "teksti",
        },
        Object {
          "data": Object {
            "rows": Array [
              Object {
                "columns": Array [
                  Object {
                    "index": 0,
                    "text": Object {
                      "fi": "Fi column1",
                      "sv": "Sv column1",
                    },
                  },
                  Object {
                    "index": 1,
                    "text": Object {
                      "fi": "Fi column2",
                      "sv": "Sv column2",
                    },
                  },
                ],
                "index": 0,
              },
              Object {
                "columns": Array [
                  Object {
                    "index": 0,
                    "text": Object {
                      "fi": "Fi column3",
                      "sv": "Sv column3",
                    },
                  },
                  Object {
                    "index": 1,
                    "text": Object {
                      "fi": "Fi column4",
                      "sv": "Sv column4",
                    },
                  },
                ],
                "index": 1,
              },
            ],
          },
          "tyyppi": "taulukko",
        },
      ],
    },
  ],
  "yhteyshenkilot": Array [
    Object {
      "nimi": Object {
        "fi": "Fi nimi",
        "sv": "Sv nimi",
      },
      "puhelinnumero": Object {
        "fi": "Fi puhelinnumero",
        "sv": "Sv puhelinnumero",
      },
      "sahkoposti": Object {
        "fi": "Fi sähköposti",
        "sv": "Sv sähköposti",
      },
      "titteli": Object {
        "fi": "Fi titteli",
        "sv": "Sv titteli",
      },
      "verkkosivu": Object {
        "fi": "Fi verkkosivu",
        "sv": "Sv verkkosivu",
      },
    },
  ],
  "ylemmanKorkeakoulututkinnonOsaamisalat": Array [
    Object {
      "kuvaus": Object {
        "fi": "<p>Fi ylempikuvaus</p>",
        "sv": "<p>Sv ylempikuvaus</p>",
      },
      "linkki": Object {
        "fi": "Fi ylempilinkki",
        "sv": "Sv ylempilinkki",
      },
      "nimi": Object {
        "fi": "Fi ylempinimi",
        "sv": "Sv ylempinimi",
      },
      "otsikko": Object {
        "fi": "Fi ylempiotsikko",
        "sv": "Sv ylempiotsikko",
      },
    },
  ],
}
`;
